File: C++/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result = {-1, -1};
        map<int, int> ref;
        
        for(int i = 0; i < nums.size(); i++)
        {
            int dif = target - nums[i];
            if(ref.find(dif) == ref.end())
                ref[nums[i]] = i;
            else
            {
                result[0] = i;
                result[1] = ref[dif];
                break;
            }
        }
        return result;
    }
};

File: Java/1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int result[] = {-1, -1};
        HashMap<Integer, Integer> ref = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++)
        {
            int dif = target - nums[i];
            if(ref.containsKey(dif))
            {
                result[0] = i;
                result[1] = ref.get(dif);
                break;
            }
            ref.put(nums[i], i);
        }
        return result;
    }
}